scope3_methodology/ad_tech_platform/model.py

```diff
@@ -0,0 +1,330 @@
+            self.get_server_allocation_decimal_fraction()
+            * self.get_server_emissions_g_co2e_per_month()
+            * self.get_servers_processing_bid_requests_decimal_fraction()
+        ) / self.get_bid_requests_processed_per_month()
+
+        log_result(
+            "server emissions g co2e per bid request",
+            f"{server_emissions_g_co2e_per_bid_request:.6f}",
+            depth,
+        )
+        return server_emissions_g_co2e_per_bid_request
+
+    def comp_primary_emissions_g_co2e_per_bid_request(
+        self,
+        depth: int,
+    ) -> float:
+        data_transfer_emissions_g_co2e_per_bid_request = (
+            self.comp_data_transfer_emissions_g_co2e_per_bid_request(depth - 1)
+        )
+
+        server_emissions_g_co2e_per_bid_request = self.comp_server_emissions_g_co2e_per_bid_request(
+            depth - 1
+        )
+        primary_emissions_g_co2e_per_bid_request = (
+            self.corporate_emissions_g_co2e_per_bid_request
+            + data_transfer_emissions_g_co2e_per_bid_request
+            + server_emissions_g_co2e_per_bid_request
+        )
+        log_result(
+            "primary emissions g co2e per bid request",
+            f"{primary_emissions_g_co2e_per_bid_request:.6f}",
+            depth,
+        )
+        return primary_emissions_g_co2e_per_bid_request
+
+    def comp_secondary_emissions_g_co2e_per_bid_request(self, depth: int) -> float:
+        secondary_emissions_per_bid_request = 0.0
+        for edge in self.distribution_partners:
+            secondary_emissions_per_bid_request += (
+                edge.partner.primary_bid_request_emissions_g_co2e
+                * edge.bid_request_distribution_rate
+            )
+        secondary_emissions_per_bid_request *= 1 - self.bid_request_rejection_rate
+        log_result(
+            "secondary emissions g co2e per bid request", secondary_emissions_per_bid_request, depth
+        )
+        return secondary_emissions_per_bid_request
+
+    def comp_cookie_syncs_processed_per_month(self, depth: int) -> float:
+        cookie_syncs_processed_per_month = self.get_cookie_syncs_processed_per_month()
+        if cookie_syncs_processed_per_month:
+            return cookie_syncs_processed_per_month
+
+        cookie_syncs_processed = (
+            self.get_bid_requests_processed_per_month()
+            * self.cookie_syncs_processed_per_bid_request
+        )
+        log_result("cookie syncs processed per month", cookie_syncs_processed, depth)
+        return cookie_syncs_processed
+
+    def comp_water_usage_to_emissions_ratio_h2o_m_3_per_g_co2e(self, depth: int) -> float:
+        water_m3_per_emissions = (
+            self.datacenter_water_intensity_h2o_m_3_per_mwh / self.server_emissions_g_per_kwh / 1000
+        )
+        log_result("h2o m^3 per g co2e emissions", water_m3_per_emissions, depth)
+        return water_m3_per_emissions
+
+    def comp_primary_emissions_g_co2e_per_cookie_sync(
```

Comment:
It's not clear to me why are we multiplying the cookie sync emissions by water usage? This doesn't then return emissions by cookie sync. 

This function is currently `get_primary_emissions_per_cookie_sync` I renamed it to included `g_co2e` so we can be clear what we are computing, but in actuality this returns `water usage per cookie sync` Seems like this should be it's own function. https://github.com/scope3data/methodology/blob/main/scope3_methodology/adtech_model.py#L254 

i.e. 
```
    def comp_primary_emissions_g_co2e_per_cookie_sync(
        self,
        depth: int,
    ) -> float:
        primary_emissions_per_cookie_sync = (
            self.get_server_allocation_decimal_fraction()
            * self.get_server_emissions_g_co2e_per_month()
            * self.get_servers_processing_cookie_syncs_decimal_fraction()
        ) / self.comp_cookie_syncs_processed_per_month(depth)

        log_result("primary emissions g per cookie sync", primary_emissions_per_cookie_sync, depth)
       return primary_emissions_per_cookie_sync

   def comp_water_usage_per_cookie_sync(
        self,
        depth: int,
    ) -> float:
        water_usage_to_emissions_ratio = (
            self.comp_water_usage_to_emissions_ratio_h2o_m_3_per_g_co2e(depth - 1)
        )
        primary_water_usage = comp_primary_emissions_g_co2e_per_cookie_sync() * water_usage_to_emissions_ratio
        log_result("primary water usage m^3 per cookie sync", primary_water_usage, depth)
        return primary_water_usage
```

Reply:
I'm sure it's copy & paste error on my part